source/kraken/maintenance_worker.cpp

```diff
@@ -73,6 +189,110 @@ void MaintenanceWorker::load_data() {
     LOG4CPLUS_INFO(logger, "Loading database duration: " << duration);
 }
 
+void MaintenanceWorker::create_task_queue() {
+    if (task_queue_created) {
+        return;
+    }
+
+    std::string instance_name = conf.instance_name();
+    std::string hostname = get_hostname();
+    std::string default_queue_name = "kraken_" + hostname + "_" + instance_name;
+    std::string queue_name = conf.broker_queue(default_queue_name);
+    queue_name_task = (boost::format("%s_task") % queue_name).str();
+
+    // first we have to delete the queues, binding can change between two run, and it's don't seem possible
+    // to unbind a queue if we don't know at what topic it's subscribed
+    // if the queue doesn't exist an exception is throw...
+    try {
+        channel->DeleteQueue(queue_name_task);
+    } catch (const std::runtime_error&) {
+    }
+    std::string exchange_name = conf.broker_exchange();
+    this->channel->DeclareExchange(exchange_name, "topic", false, true, false);
+
+    // creation of task queue for this kraken
+    bool passive = false;
+    bool durable = true;
+    bool exclusive = false;
+    bool auto_delete_queue = conf.broker_queue_auto_delete();
+
+    AmqpClient::Table args;
+    args.insert(std::make_pair("x-expires", conf.broker_queue_ttl() * 1000));
```

Comment:
The doc is here: https://www.rabbitmq.com/ttl.html#queue-ttl
There are things that keep the queue "alive": consumer and re-declaration.
New messages don't, and that's what we want here: remove unconsumed queues that pile up messages and block all the Rabbit.

I'm not sure for "task" but for "rt", we can remove the queue as soon as the kraken linked to it is restarted (it will recreate the "rt" queue and ask for previously known realtime.
The queue should only be kept if kraken is still alive and can later reconnect.